eth/gasprice/gasprice.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2015 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package gasprice
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	cm "github.com/celo-org/celo-blockchain/contracts/currency"
+	gpm "github.com/celo-org/celo-blockchain/contracts/gasprice_minimum"
+	"github.com/celo-org/celo-blockchain/core/vm"
+	"github.com/celo-org/celo-blockchain/params"
+)
+
+var (
+	suggestionMultiplier *big.Int = big.NewInt(5) // The multiplier that we apply to the minimum when suggesting gas price
+)
+
+func GetBaseFeeForCurrency(vmRunner vm.EVMRunner, currencyAddress *common.Address, baseFee *big.Int) (*big.Int, error) {
+	// Gingerbread Fork
+	if baseFee != nil {
+		if currencyAddress == nil {
+			return baseFee, nil
+		} else {
+			exchangeRate, err := cm.GetExchangeRate(vmRunner, currencyAddress)
+			if err != nil {
+				// Assign zero if the exchangeRate fails to mimic the same behaviour as Espresso
+				return big.NewInt(0), err
```

Comment:
Is zero a good fallback here? Shouldn't we make transactions expensive rather than free if in doubt?

Reply:
Yes, haha, agree on that. I've added it here as zero because it was the same behaviour we had on production. Technically speaking, if we don't have the gasPriceMinimum contract nor the sorted oracles of a stable, we are probably in a bigger problem, but yes, we could change the default on how we handle this

Reply:
I agree that his is a rare edge case and the specific handling is not too important. We can leave it as zero to minimize the changes for now, if you think that is safer.